### PR TITLE
fix: [MAGIC] Several fixes 🐛 

### PR DESCRIPTION
**Pull request by ahmdyasser**
#### Proposed Changes
As title

#### Screenshot
| Before | After |
| --- | ----------- |
|||

#### Checklist

- [x] I have tested these changes locally.
- [ ] I have added appropriate documentation or updated existing documentation.
- [ ] I have added appropriate test coverage or updated existing test coverage.
- [ ] I have updated the changelog (if applicable).
- [x] I have followed the project's code style and formatting guidelines.
- [x] I have reviewed and adhered to the project's contributing guidelines.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -92,4 +92,5 @@ iOSInjectionProject/
 
 # System files
 
-.DS_Store
\ No newline at end of file
+.DS_Store
+Healthy/Resources/Generated
```
Comment (ahmdmhasn) [inline .gitignore]:
```suggestion

# Ignore generated code
*/*/Generated
```

Not sure if this will work but it's better than using specific destination.

Comment (ahmdmhasn) [reply .gitignore]: Also, we need to consider regeneration of the files in the workflow, it fails 👀

File: Healthy/Classes/Extensions/UIView+Style.swift [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        var radius: CGFloat
+        var color: CGColor
+        var opacity: Float
+    }
+
+    func applyShadow(_ type: ShadowType) {
+        switch type {
+        case .normal:
+            let offset = CGSize(width: 0, height: 0)
+            let color = UIColor(red: 0.412, green: 0.412, blue: 0.412, alpha: 0.1).cgColor
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIView+Style.swift]: Shall we use the colors in the color scheme?

File: Healthy/Classes/Extensions/UIView+Style.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            let style = UIButton.ShadowStyle(
```
Comment (ahmdmhasn) [inline Healthy/Classes/Extensions/UIView+Style.swift]:
```suggestion
            let style = ShadowStyle(
```

Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UIView+Style.swift]: Why create a struct to hold values and then re-use the struct to apply those values again?

File: Healthy/Classes/Modules/Login/LoginViewController.swift [hunk 4]
```diff
@@ -73,6 +75,13 @@ private extension LoginViewController {
         passwordTextFieldLabel.applyStyle(.textFieldTitleLabel)
         signInButton.applyButtonStyle(.primary)
         signInWithFacebookButton.applyButtonStyle(.plainGold)
+        signInWithGoogleButton.applyShadow(.normal)
+        signInWithFacebookButton.applyShadow(.normal)
+    }
+
+    func configureContent() {
+        signInWithGoogleButton.setImage(.iconGoogle, for: .normal)
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Login/LoginViewController.swift]: Thanks for making this!

File: Healthy/Classes/Modules/Login/LoginViewController.xib [hunk 5]
```diff
@@ -124,33 +125,31 @@
...
                                             <inset key="imageEdgeInsets" minX="0.0" minY="0.0" maxX="2.2250738585072014e-308" maxY="0.0"/>
-                                            <state key="normal" image="die.face.5.fill" catalog="system"/>
+                                            <buttonConfiguration key="configuration" style="plain"/>
                                             <connections>
-                                                <action selector="didTapSignInWithFacebook" destination="-1" eventType="touchUpInside" id="C79-aL-XIL"/>
+                                                <action selector="didTapSignInWithFacebook:" destination="-1" eventType="touchUpInside" id="S48-V5-DAK"/>
                                             </connections>
                                         </button>
                                     </subviews>
+                                    <constraints>
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Login/LoginViewController.xib]: Using a horizontal stack view with equal width alignment and add this to a parent stack with center distribution would serve better. You will need to add only one constraint for a button and when it comes to add new Sign-in with Apple, we won't need to add and new constraint.

File: Healthy/Classes/Modules/Splash/SplashViewController.swift [hunk 6]
```diff
@@ -30,6 +30,7 @@ final class SplashViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
 
+        configureContent()
```
Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Splash/SplashViewController.swift]: 👏🏼

File: Healthy/Resources/Assets.xcassets/right-arrow.imageset/Contents.json
Comment (ahmdmhasn) [inline Healthy/Resources/Assets.xcassets/right-arrow.imageset/Contents.json:1]: Please name to **icon-right-arrow** to follow the convention.

File: .gitignore (hunk 1 above)
Comment (ahmdmhasn) [reply .gitignore]: This might be the most important change in the PR... LMK if you need any assistance.

**Review by ahmdmhasn — COMMENTED** [review]
fix: [MAGIC] Several fixes 🐛


Thanks for taking time to add those changes, much appreciated! ❤️

File: Healthy/Classes/Extensions/UIView+Style.swift (hunk 2 above)
Comment (ahmdyasser) [reply Healthy/Classes/Extensions/UIView+Style.swift]: Am not sure, the shadow color in Figma design is in hex without a design system color name. What do you think?

File: .gitignore (hunk 1 above)
Comment (ahmdyasser) [reply .gitignore]: I'll try that wildcard pattern. 

Let the workflow issue to me, I'll handle that.

File: Healthy/Classes/Extensions/UIView+Style.swift (hunk 3 above)
Comment (ahmdyasser) [reply Healthy/Classes/Extensions/UIView+Style.swift]: Yes, forgot to remove it.

File: Healthy/Classes/Extensions/UIView+Style.swift (hunk 2 above)
Comment (ahmdmhasn) [reply Healthy/Classes/Extensions/UIView+Style.swift]: Choose gray40 ... The closest to this color code

File: Healthy/Resources/Localization/Localizable.strings [hunk 7]
```diff
@@ -5,3 +5,9 @@
 // MARK: Common Localizations
 "common.email" = "Email";
 "common.password" = "Password";
+
+// MARK: Splash Screen
+"splash.main.title" = "Get\nCooking";
+"splash.main.subtitle" = "Simple way to find Tasty Recipe";
+"splash.button.title" = "Start Cooking";
```
Comment (ahmdmhasn) [inline Healthy/Resources/Localization/Localizable.strings]:
```suggestion
"splash.login.button.title" = "Start Cooking";
```

**Pull request closed without merging**